rtc_service/rtc_service.py

```diff
@@ -0,0 +1,288 @@
+        sink_manager
+    ):
+        """
+            Thread sending periodically time to synchronize the network.
+
+        Args:
+            period: The period to send gateway time to the network
+            sink_manager: The sink manager to send sink the rtc informations
+        """
+        Thread.__init__(self)
+
+        # Daemonize thread to exit with full process
+        self.daemon = True
+
+        # How often to send time
+        self.period = period
+        self.timezone_from_gateway_clock = timezone_from_gateway_clock
+        if timezone_from_gateway_clock is True:
+            self.timezone_offset_s = datetime.datetime.now(datetime.timezone.utc).astimezone().utcoffset().seconds
+            logging.info("Timezone offset is initialized with gateway clock")
+        else:
+            self.timezone_offset_s = timezone_offset_s
+            logging.info("Timezone offset is initialized with parameter")
+        logging.info(f"Timezone offset is set to {self.timezone_offset_s}s")
+
+        self.get_time_from_local = get_time_from_local
+        if not self.get_time_from_local:
+            logging.info("RTC time is taken from a ntp server")
+            self.ntp_client = ntplib.NTPClient()
+        else:
+            logging.info("RTC time is taken from local time")
+        self.sink_manager = sink_manager
+        logging.info(f"Expected RTC sending period is set to {self.period}s")
+
+        self.running = False
+
+    def publish_time(self):
+        """
+        Publish the rtc time in the network.
+        """
+        if not self.get_time_from_local:
+            try:
+                req = self.ntp_client.request('pool.ntp.org', version=3)
+                timestamp = req.dest_time+req.offset
+            except ntplib.NTPException as err:
+                logging.warning("Couldn't get time from NTP server. (%s)", err)
+                return
+        else:
+            timestamp = time()
+
+        timestamp_ms = int(timestamp*1000)
+        data_payload = RTC_VERSION.to_bytes(2, "little") + encode_tlv(timestamp_ms, self.timezone_offset_s)
+        logging.info("Send rtc message to the network")
+        for sink in self.sink_manager.get_sinks():
+            start = time()
+            sink.send_data(
+                dst=BROADCAST_ADDRESS,
+                src_ep=RTC_SOURCE_EP,
+                dst_ep=RTC_DEST_EP,
+                qos=0,
+                initial_time=INITIAL_DELAY_MS,
+                data=data_payload,
+                is_unack_csma_ca=False,
+                hop_limit=0,
+            )
+            logging.debug("time elapsed to send RTC time through the gateway: "
+                          f"{int((time()-start)*1000)}ms")
+
+    def run(self):
+        """
+        Main loop that send periodically gateway time to the network.
+        """
+        self.running = True
+
+        while self.running:
+            self.publish_time()
+            sleep(self.period)
+
+    def stop(self):
```

Comment:
It looks like you never call stop.
Is it needed to keep it?

Thread is daemonized so should be killed when the process is closed

Reply:
I'm going to remove this function. 
stop() function in the transport service is also never called and it's also daemonized. Should we remove this one as well ?